### PR TITLE
Trying isomorphic styles and CSS modules

### PR DESCRIPTION
**Pull request by scruffian**
In get.blog we take a different approach to CSS from Calypso. Rather than carefully name spacing each class name, we use an isomorphic style loader to generate our CSS on the fly. This gives us access to [CSS modules](https://github.com/css-modules/css-modules), which means we don't need to worry about name space collisions. It also means we only load the CSS for each section when we need it.

This PR makes an attempt to add isomorphic styles to Calypso, but it doesn't solve the following problems:

1. How should this work with SSR?
2. How can we include the `<Stylizer>` component on every path without explicitly declaring it each time.
3. What impact will moving to Webpack 2 have?
4. Should we make Stylizer a separate package on npm?

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -37,6 +38,9 @@ const webpackConfig = {
 		chunkFilename: '[name].[chunkhash].js', // ditto
 		devtoolModuleFilenameTemplate: 'app:///[resource-path]'
 	},
+	postcss() {
```
Comment (aduth) [inline webpack.config.js]: FYI this is invalid in more recent versions of Webpack 2.x which implement strict config validation. Instead you'll probably want to create a `.postcssrc` file.

File: client/components/vertical-nav/index.jsx [hunk 2]
```diff
@@ -2,15 +2,17 @@
...
  */
 import React from 'react';
+import style from './style';
+import withStyles from 'isomorphic-style-loader/lib/withStyles';
 
 const VerticalNav = React.createClass( {
 	render() {
 		return (
-			<div className="vertical-nav">
+			<div className={ style.verticalNav }>
```
Comment (aduth) [inline client/components/vertical-nav/index.jsx]: How is `verticalNav` defined? Is there meant to be a `style.scss` file included in this branch? Curious how this looks from a stylesheet writing point-of-view.

Comment (scruffian) [inline client/components/vertical-nav/index.jsx]: This uses the stylesheet that is already there for vertical-nav - that's one of the advantages of this approach - we don't need to change any of the CSS files to convert components.

Comment (aduth) [inline client/components/vertical-nav/index.jsx]: Ah, so it creates an object of selectors it discovers in `style.scss` and camel-cases them as properties?

Comment (scruffian) [inline client/components/vertical-nav/index.jsx]: Yep that's right

File: webpack.config.js (hunk 1 above)
Comment (yurynix) [inline webpack.config.js]: Yep, we migrated that in Delphin:
https://github.com/Automattic/delphin/blob/master/webpack.base.config.js#L64

Comment (yurynix) [inline webpack.config.js]: It'll maybe useful to take a look at #665-gh-delphin where a similar issue is addressed

File: package.json [hunk 3]
```diff
@@ -130,7 +134,7 @@
     "uuid": "2.0.1",
     "valid-url": "1.0.9",
     "walk": "2.3.4",
-    "webpack": "1.13.1",
+    "webpack": "2.1.0-beta.25",
```
Comment (blowery) [inline package.json]: huh, so webpack2 just works? Maybe we should try to land that in a separate PR?

Comment (scruffian) [inline package.json]: It seems to work, but I didn't test it carefully. We should probably upgrade even if we don't decide to use this approach.

**Comment by gziolo** [comment]
Unfortunately this PR doesn’t have `shrinkwrap` file regenerated and it’s still using Webpack 1.x.

I tried to regenerated `shrinkwrap` file, but it didn't work even when I updated the following packages to the latest versions:
* "hard-source-webpack-plugin": "0.3.6"
* "webpack": "2.2.0-rc.4"
* "webpack-dev-middleware": "1.9.0"
* "webpack-dev-server": "2.2.0-rc.0"

It looks like this is enough to perform successful run of `make shrinkwrap`, but when I try to execute `make run` I get the following error:

> Error: Cannot find module 'webpack/lib/dependencies/NullDependencyTemplate'
>     at Function.Module._resolveFilename (module.js:469:15)
>     at Function.Module._load (module.js:417:25)
>     at Module.require (module.js:497:17)
>     at require (internal/module.js:20:19)
>     at Object.<anonymous> (/Users/gziolo/PhpstormProjects/wp-calypso/node_modules/hard-source-webpack-plugin/index.js:34:30)
>     at Module._compile (module.js:570:32)
>     at Object.Module._extensions..js (module.js:579:10)
>     at Module.load (module.js:487:32)
>     at tryModuleLoad (module.js:446:12)
>     at Function.Module._load (module.js:438:3)
>     at Module.require (module.js:497:17)
>     at require (internal/module.js:20:19)
>     at Object.<anonymous> (/Users/gziolo/PhpstormProjects/wp-calypso/webpack.config.node.js:8:28)
>     at Module._compile (module.js:570:32)
>     at Object.Module._extensions..js (module.js:579:10)
>     at Module.load (module.js:487:32)
> make: *** [build-server] Error 1
>

File: package.json (hunk 3 above)
Comment (gziolo) [inline package.json]: Shrinkwrap wasn't regenerated so this patch is still using old Webpack version ;)

**Comment by stephanethomas** [comment]
I got CSS modules working in development with the current version of Webpack in https://github.com/Automattic/wp-calypso/pull/10590.
